pkg/services/ngalert/tests/fakes/rules.go

```diff
@@ -342,3 +342,37 @@ func (f *RuleStore) IncreaseVersionForAllRulesInNamespace(_ context.Context, org
 func (f *RuleStore) CountInFolders(ctx context.Context, orgID int64, folderUIDs []string, u identity.Requester) (int64, error) {
 	return 0, nil
 }
+
+func (f *RuleStore) GetOrCreateDefaultAlertingNamespace(ctx context.Context, orgID int64) (*folder.Folder, error) {
```

Comment:
once file provisioning user is created, we can make this method accept the user and we will not have to create fake user here. Also, this will protect us from possible future problems in provisioning API (e.g. due to a bug user will be able to create folders via rules API)